src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -93,6 +93,7 @@ public void MultipleTimers_ShouldFire_InCorrectOrder()
             if (count == 0)
             {
                 Assert.Same(t2State, o);
+                Interlocked.Increment(ref count);
```

Comment:
Additionally, GitHub won't let me comment on them because they weren't changed as part of this PR, but I just noticed that the PassesStateThroughCallback tests are likely not testing what you want them to be testing, because you're constructing the timer and then immediately disposing it... there's a good chance it'll be disposed before the callback gets invoked, at which point we'll never run the asserts in the callback.